File: leetcode/AIcoding/4.GPT-4o/c_solutions/2350-shortest-impossible-sequence-of-rolls.c
int shortestImpossibleSequence(int* rolls, int rollsSize) {
    int count[7] = {0};
    for (int i = 0; i < rollsSize; i++) {
        count[rolls[i]]++;
    }

    for (int length = 1; length <= 6; length++) {
        int total = 0;
        for (int i = 1; i <= 6; i++) {
            total += count[i] > 0 ? 1 : 0;
            if (total == length) {
                break;
            }
        }
        if (total < length) {
            return length;
        }
    }
    return 7;
}

File: leetcode/AIcoding/1.Gemini-2.5/erlang_solutions/2350-shortest-impossible-sequence-of-rolls.erl
-module(solution).
-export([shortestImpossibleSequence/2]).

shortestImpossibleSequence(Rolls, K) ->
    {_CurrentNum, NumSequences} = lists:foldl(
        fun(Roll, {CurrentNum, AccNumSequences}) ->
            if
                Roll == CurrentNum ->
                    NewCurrentNum = CurrentNum + 1,
                    if
                        NewCurrentNum > K ->
                            {1, AccNumSequences + 1};
                        true ->
                            {NewCurrentNum, AccNumSequences}
                    end;
                true ->
                    {CurrentNum, AccNumSequences}
            end
        end,
        {1, 0},
        Rolls
    ),
    NumSequences + 1.